ddtrace/debugging/_probe/registry.py

```diff
@@ -0,0 +1,185 @@
+from collections import defaultdict
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+
+from ddtrace.debugging._probe.model import FunctionProbe
+from ddtrace.debugging._probe.model import LineProbe
+from ddtrace.debugging._probe.model import Probe
+from ddtrace.debugging._probe.status import ProbeStatusLogger
+from ddtrace.internal import forksafe
+from ddtrace.internal.compat import ExcInfoType
+from ddtrace.internal.logger import get_logger
+
+
+logger = get_logger(__name__)
+
+
+class ProbeRegistryEntry(object):
+
+    __slots__ = ("probe", "installed", "exc_info", "message")
+
+    def __init__(self, probe):
+        # type: (Probe) -> None
+        self.probe = probe
+        self.installed = False
+        self.exc_info = None  # type: Optional[ExcInfoType]
+        self.message = None  # type: Optional[str]
+
+    def activate(self):
+        # type: () -> None
+        return self.probe.activate()
+
+    def deactivate(self):
+        # type: () -> None
+        return self.probe.deactivate()
+
+    def set_installed(self):
+        # type: () -> None
+        self.installed = True
+
+    def set_exc_info(self, exc_info):
+        # type: (ExcInfoType) -> None
+        self.exc_info = exc_info
+
+    def set_message(self, message):
+        # type: (str) -> None
+        self.message = message
+
+
+def _get_probe_location(probe):
+    # type: (Probe) -> Optional[str]
+    if isinstance(probe, LineProbe):
+        return probe.source_file
+    elif isinstance(probe, FunctionProbe):
+        return probe.module
+    else:
+        raise ValueError("Unsupported probe type: {}".format(type(probe)))
```

Comment:
Do we worry about ever crashing a customer's application if we hit this? Should we catch this somewhere and write an error log?

Reply:
The worst that can happen is that we crash the debugger. This is there as a "safety net" but we expect to never hit this line.